hare/eligibility/oracle.go

```diff
@@ -344,11 +338,16 @@ func (o *Oracle) CalcEligibility(
 
 // Proof returns the role proof for the current Layer & Round.
 func (o *Oracle) Proof(ctx context.Context, layer types.LayerID, round uint32) (types.VrfSignature, error) {
-	msg, err := o.buildVRFMessage(ctx, layer, round)
+	beacon, err := o.beacons.GetBeacon(layer.GetEpoch())
 	if err != nil {
-		return types.EmptyVrfSignature, err
+		return types.EmptyVrfSignature, fmt.Errorf("get beacon: %w", err)
 	}
-	return o.vrfSigner.Sign(msg), nil
+	return o.GenVRF(ctx, o.vrfSigner, beacon, layer, round), nil
+}
```

Comment:
`buildVRFMessage` and `GenVRF` do essentially the same. Are both needed? This could also be written as:

```go
	msg, err := o.buildVRFMessage(ctx, layer, round)
	if err != nil {
		return types.EmptyVrfSignature, err
	}
	return msg, nil
```

Reply:
it is used in code path for hare1, don't want to spend time refactoring that